Library/Homebrew/commands.rb

```diff
@@ -197,4 +197,8 @@ def rebuild_commands_completion_list
     file = HOMEBREW_CACHE/"all_commands_list.txt"
     file.atomic_write("#{commands(aliases: true).sort.join("\n")}\n")
   end
+
+  def only_bash_command_list
+    internal_commands.reject { |cmd| valid_internal_cmd?(cmd) }
```

Comment:
I think this function makes sense but not the implementation. This should check for a file extension instead.